default.nix

```diff
@@ -3,24 +3,30 @@
 , system ? builtins.currentSystem
 
 , haskellNix ? import
-    (builtins.fetchTarball
-      "https://github.com/input-output-hk/haskell.nix/archive/0.0.64.tar.gz")
-    { }
+    (builtins.fetchTarball {
+      url = "https://github.com/input-output-hk/haskell.nix/archive/0.0.64.tar.gz";
+      sha256 = "0mply6n3gym8r7z4l7512phw667qzwfqkfl1zmiyrdjpdg7xqn1d";
+    })
+    { pkgs = import nixpkgsSrc { inherit system; }; }
```

Comment:
We need to pass in the system so that `import nixpkgsSrc` does not default to `builtins.currentSystem`, which is not available in pure evaluation. Pure evaluation is desirable for caching and reproducibility reasons.